### PR TITLE
feat: add annotations to schema

### PR DESCRIPTION
**Pull request by sissbruecker**
## Description

Adds annotations for properties to the schema. The annotations are filtered by an include list, which currently only includes `jakarta.persistence.Id` and `jakarta.persistence.Version`.

Part of https://github.com/vaadin/hilla/issues/1266

### CODE REVIEW

File: packages/java/parser-jvm-plugin-model/src/main/java/dev/hilla/parser/plugins/model/ModelPlugin.java [hunk 1]
```diff
@@ -53,20 +72,31 @@ private static boolean isValidationConstraintAnnotation(
...
 
+    private static boolean isIncludedAnnotation(
+            AnnotationInfoModel annotation) {
+        return INCLUDED_ANNOTATIONS.contains(annotation.getName());
+    }
+
     @Override
     public void enter(NodePath<?> nodePath) {
-        if (!(nodePath.getNode() instanceof TypedNode)) {
+        if (!(nodePath.getNode()instanceof TypedNode typedNode)) {
```
Comment (Lodin) [inline packages/java/parser-jvm-plugin-model/src/main/java/dev/hilla/parser/plugins/model/ModelPlugin.java:82]:
```suggestion
        if (!(nodePath.getNode() instanceof TypedNode typedNode)) {
```

File: packages/java/parser-jvm-plugin-model/src/main/java/dev/hilla/parser/plugins/model/ModelPlugin.java [hunk 2]
```diff
@@ -53,20 +72,31 @@ private static boolean isValidationConstraintAnnotation(
...
             return;
         }
 
         var schema = typedNode.getTarget();
-        addConstraintsToSchema((AnnotatedNode) typedNode, schema);
+        addConstraintsToSchema(typedNode, schema);
+
+        // Add annotations from parent property model to schema
+        if (nodePath.getParentPath() != null && nodePath.getParentPath()
+                .getNode()instanceof PropertyNode propertyNode) {
```
Comment (Lodin) [inline packages/java/parser-jvm-plugin-model/src/main/java/dev/hilla/parser/plugins/model/ModelPlugin.java:96]:
```suggestion
                .getNode() instanceof PropertyNode propertyNode) {
```

**Review by Lodin — CHANGES_REQUESTED** [review]
There are some stylistic issues to fix

File: packages/java/parser-jvm-plugin-model/src/main/java/dev/hilla/parser/plugins/model/ModelPlugin.java (hunk 1 above)
Comment (sissbruecker) [reply packages/java/parser-jvm-plugin-model/src/main/java/dev/hilla/parser/plugins/model/ModelPlugin.java:82]: Unfortunately that is how `mvn formatter:format` decides to format this.

Comment (Lodin) [reply packages/java/parser-jvm-plugin-model/src/main/java/dev/hilla/parser/plugins/model/ModelPlugin.java:82]: That's weird. Probably, they don't know what to do with that new syntax. But ok, let's have it that way then.

**Review by Lodin — DISMISSED** [review]

**Comment by Artur-** [comment]
Now you broke react-grid. Probably something with the Filter class and its annotations

File: packages/java/parser-jvm-plugin-model/src/main/java/dev/hilla/parser/plugins/model/ModelPlugin.java (hunk 1 above)
Comment (Artur-) [reply packages/java/parser-jvm-plugin-model/src/main/java/dev/hilla/parser/plugins/model/ModelPlugin.java]: Oh look it was resolved with the latest formatter :)

**Review by Artur- — APPROVED** [review]

**Pull request merged**